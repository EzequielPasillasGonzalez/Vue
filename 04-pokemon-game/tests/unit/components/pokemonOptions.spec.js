import { shallowMount } from "@vue/test-utils";
import PokemonOptions from "@/components/PokemonOptions.vue";
import { mockPokemons } from "../mocks/pokemons.mock";

describe("PokemonOptions componente", () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallowMount(PokemonOptions, {
      props: {
        pokemons: mockPokemons,
      },
    });
    console.log(wrapper.html());
  });

  test("Debe de hacer match con el snapshot", () => {

    expect(wrapper.html()).toMatchSnapshot();
  });

  test('Debe de mostrar las 4 opciones correctamente', () => { 
      // Que existan 4 li
      // Cada li debe de tener el nombre

      const liTags = wrapper.findAll('li')

      expect(liTags.length).toBe(mockPokemons.length); //? Muestra si existen
      
      for (let index = 0; index < mockPokemons.length; index++) {
        expect(liTags[index].text()).toBe(mockPokemons[index].name); //? Revisan que tengan su el nombre de los pokemon   
      }                        


  })

  test('Debe de emitir "selection" con sus respectivos parametros al hacer click', () => { 

      const liTags = wrapper.findAll('li')            
            
      for (let index = 0; index < liTags.length; index++) {        
          
          liTags[index].trigger('click') //? emite el even to click en todos los li          
        
          expect(wrapper.emitted('selection')[index]).toEqual([mockPokemons[index].id]) //? Se espera que los id que emiten coincidan
      }           

      expect(wrapper.emitted('selection').length).toBe(4) //? se espera que se emitieran 4 eventos      

  })
});
