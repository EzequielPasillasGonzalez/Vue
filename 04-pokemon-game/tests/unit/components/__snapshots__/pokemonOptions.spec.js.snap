// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PokemonOptions componente Debe de hacer match con el snapshot 1`] = `
<div class="options-container">
  <ul>
    <li>
      <!--? Con $emit es para nombrar el evento, primero se manda el nombre del evento -->
      <!--? Despues se manda el argumento que se quiera  --> bulbasaur
    </li>
    <li>
      <!--? Con $emit es para nombrar el evento, primero se manda el nombre del evento -->
      <!--? Despues se manda el argumento que se quiera  --> ivysaur
    </li>
    <li>
      <!--? Con $emit es para nombrar el evento, primero se manda el nombre del evento -->
      <!--? Despues se manda el argumento que se quiera  --> venusaur
    </li>
    <li>
      <!--? Con $emit es para nombrar el evento, primero se manda el nombre del evento -->
      <!--? Despues se manda el argumento que se quiera  --> charmander
    </li>
  </ul>
</div>
`;
